### PR TITLE
EX2; compile and run success but uncertainly work correct

### PR DESCRIPTION
**Pull request by tiennma2k23**

### CODE REVIEW

**Review by Sukkaito — COMMENTED** [review]
Please separate each smaller questions into different commits next time :(

File: src/App.cpp [hunk 1]
```diff
@@ -573,6 +573,60 @@ void createAGVs()
                 }
                 socialForce->addAGV(agv);
 
+                int marker = numRunPerHallway * (juncIndexTemp + 1) - 1;
+                if ((int)inputData["runConcurrently"]["value"] == 1)
```
Comment (Sukkaito) [inline src/App.cpp:577]: Already check at Line 543

**Comment by tiennma2k23** [comment]
> Please separate each smaller questions into different commits next time :(

okayyy. Please add other members. I need them to the review

File: data/input.json [hunk 2]
```diff
@@ -86,5 +86,22 @@
...
+  "randomness": {
+    "description": "the option for establishing new randon configuration or load from the file",
+    "value": 0 
+  },
+  "ageDistribution" : {
+    "description" : "distribution of age in pedestrians",
+    "distribution": {
+      "normal" : {
+        "description" : "normal distribution",
+        "samples" : 50,
```
Comment (Sukkaito) [inline data/input.json]: Just leave `samples` field as "numOfAgents", we don't wanna change it everytime we change the sample size. Just call for "numOfAgents" when you wanna use in code.

File: src/agv/AGV.cpp
Comment (Sukkaito) [inline src/agv/AGV.cpp:1]: AGV won't need randomness. Randomness only affects persons.

File: data/input.json (hunk 2 above)
Comment (Sukkaito) [reply data/input.json]: Or make it `{"$ref": "#/numOfAgents/value"}`

File: src/App.cpp [hunk 3]
```diff
@@ -573,6 +573,60 @@ void createAGVs()
...
+                        juncIndexTemp = 0;
+                    }
+                    hallwayLength = juncDataList[juncIndexTemp].items().begin().value();
+                    length1Side = (hallwayLength) / 2;
+                    juncDataTemp = {length1Side, length1Side};
+                }
+            }
+        }
+    }
+    else {
```
Comment (Sukkaito) [inline src/App.cpp:597]: runMode == 3

File: src/App.cpp [hunk 4]
```diff
@@ -573,6 +575,59 @@ void createAGVs()
                 }
                 socialForce->addAGV(agv);
 
+                int marker = numRunPerHallway * (juncIndexTemp + 1) - 1;
```
Comment (Sukkaito) [inline src/App.cpp:578]: I believe the program itself in runMode == 1 doesn't require to change anything. Am I missing something here?

Comment (Sukkaito) [reply src/App.cpp:578]: Ah never mind, it's the exact same thing in the runMode == 3. Nothing has been changed.

File: src/agv/AGV.h
Comment (Sukkaito) [inline src/agv/AGV.h:1]: AGV wont's have these properties, as it's Pedestrian properties, and only used in data generation.

File: src/App.cpp
Comment (Sukkaito) [inline src/App.cpp:1]: Age Distribution has no impact on AGVs, only individual pedestrian does.

File: src/agent/Agent.cpp
Comment (Sukkaito) [inline src/agent/Agent.cpp:1]: I think we'll have to create Pedestrian.h and Pedestrian.cpp, so no need to make changes to Agent. (Check Ex 3i for more info)

File: src/App.cpp
Comment (Sukkaito) [inline src/App.cpp:1]: Same apply here, we'll have to create new Pedestrian.h and Pedestrian.cpp. Please revert changes made to Agents.

File: src/App.cpp (hunk 3 above)
Comment (tiennma2k23) [reply src/App.cpp:597]: I think it's not necessary because it has 3 cases: 0,1 or 3

File: src/App.cpp (hunk 4 above)
Comment (tiennma2k23) [reply src/App.cpp:578]: yebb. Your comment has confused me !

**Review by Sukkaito — APPROVED** [review]
Probably good to go! Great job!

**Review by GnuTLam — APPROVED** [review]

**Pull request merged**